rllib/policy/rnn_sequencing.py

```diff
@@ -260,32 +260,34 @@ def chop_into_sequences(*,
             seq_lens.append(seq_len)
         seq_lens = np.array(seq_lens, dtype=np.int32)
 
-    assert sum(seq_lens) == len(feature_columns[0])
-
     # Dynamically shrink max len as needed to optimize memory usage
     if dynamic_max:
         max_seq_len = max(seq_lens) + _extra_padding
 
     feature_sequences = []
-    for f in feature_columns:
-        # Save unnecessary copy.
-        if not isinstance(f, np.ndarray):
-            f = np.array(f)
-        length = len(seq_lens) * max_seq_len
-        if f.dtype == np.object or f.dtype.type is np.str_:
-            f_pad = [None] * length
-        else:
-            # Make sure type doesn't change.
-            f_pad = np.zeros((length, ) + np.shape(f)[1:], dtype=f.dtype)
-        seq_base = 0
-        i = 0
-        for len_ in seq_lens:
-            for seq_offset in range(len_):
-                f_pad[seq_base + seq_offset] = f[i]
-                i += 1
-            seq_base += max_seq_len
-        assert i == len(f), f
-        feature_sequences.append(f_pad)
+    for col in feature_columns:
+        feature_sequences.append([])
+
+        for f in tree.flatten(col):
```

Comment:
Good point, will do.
I added a test case for `handle_nested_data=True`.
Otherwise, the `test_lstm` tests make sure behavior doesn't change and everything is backward compatible w/o the user having to change anything.